### PR TITLE
[#1] Add a description for each Extension

### PR DESCRIPTION
**Pull request by smcvb**
This pull request introduces short descriptions of each extension within the Axon Framework project.

More concretely, it introduces a section for the:

- AMQP extension
- JGroups extension
- JobRunrPro extension
- Kafka extension
- Kotlin extension
- Mongo extension
- Multitenancy extension
- Reactor extension
- Spring Cloud extension
- Spring Native extension
- Tracing extension

Lastly, it also moves the extensions contents section from the introduction to the dedicated Extensions section.

In providing the above, this pull request partially resolves issue #1.

### CODE REVIEW

File: profile/README.md [hunk 1]
```diff
@@ -104,28 +93,181 @@ The extensions are intentionally split off from the main framework to not over e
...
+ * **[Kotlin](#kotlin)** - Extension enhancing the development experience when using [Kotlin](https://kotlinlang.org/)
+ * **[MongoDB](#mongodb)** - Extension adding [MongoDB](https://www.mongodb.com/) integration for all Axon Framework components requiring storage
+ * **[Multitenancy](#multitenancy)** - Extension adding building blocks to simplify [multitenancy](https://en.wikipedia.org/wiki/Multitenancy), particularly straightforward in combination with [Axon Server](https://developer.axoniq.io/axon-server/overview)
+ * **[Reactor](#reactor)** - Extension providing Axon Framework-specific infrastructure components using the [Project Reactor](https://projectreactor.io/) API 
+ * **[Spring Cloud](#spring-cloud)** - Extension adding [Spring Cloud](https://spring.io/projects/spring-cloud) integration for command routing
+ * **[Spring Native](#spring-native)** - Experimental extension providing native compilation for Axon and Spring-based application through the [Spring Native](https://github.com/spring-attic/spring-native) project.  
+ * **[Tracing](#tracing)** - Extension adding [Open Tracing](https://opentracing.io/) integration to Axon's infrastructure components
```
Comment (CodeDrivenMitch) [inline profile/README.md:108]: Maybe leave this one out, as we have deprecated it? Or at least show a deprecation warning

File: profile/README.md [hunk 2]
```diff
@@ -104,28 +93,181 @@ The extensions are intentionally split off from the main framework to not over e
...
+
 ### AMQP
 
+The [AMQP Extension](https://github.com/AxonFramework/extension-amqp) repository provides support for the [Advanced Message Queuing Protocol](https://en.wikipedia.org/wiki/Advanced_Message_Queuing_Protocol) (AMQP).
+
+Users can add this extension to, for example, include a [RabbitMQ](https://www.rabbitmq.com/) integration with their Axon Framework application.
+Setting up this integration allows you to publish Axon events onto and read from a queue.
```
Comment (CodeDrivenMitch) [inline profile/README.md:115]:
```suggestion
Setting up this integration allows you to publish Axon events to an AQMP exchange. Events can also be read from exchanges and handled by event processors, providing means of integrating with other services.
```

File: profile/README.md [hunk 3]
```diff
@@ -104,28 +93,181 @@ The extensions are intentionally split off from the main framework to not over e
...
+Please keep that in mind when using this extension.
+
 ### Tracing
 
+The [Tracing Extension](https://github.com/AxonFramework/extension-tracing) repository enables integration with [Open Tracing](https://opentracing.io/).
+
+When using this extension, you will replace the `Command-` and `QueryGateway` with implementations that attach span information to your `Messages`.
+Furthermore, when handling a `Message`, a dedicated `MessageHandlerInterceptor` ensures the span is populated on the overall trace.
+
+Although this is valuable, it is essential to mention that the Open Tracing implementation has been archived in favor of [Open Telemetry](https://opentelemetry.io/).
```
Comment (CodeDrivenMitch) [inline profile/README.md:261]: Ah, there is a deprecation notice

**Review by CodeDrivenMitch — APPROVED** [review]
Looks good. Only some small things to consider

File: profile/README.md (hunk 1 above)
Comment (smcvb) [reply profile/README.md:108]: I'll mention it is superseded by the integrated OpenTelemetry support :+1:

**Pull request merged**